### PR TITLE
Readme updates

### PR DESCRIPTION
**Pull request by sbuliarca**

### CODE REVIEW

File: Pipeline_integration_points.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+### AWS Helm repo integration
+We're keeping all the UPP helm charts in an S3 bucket exposed as an HTTP static website restricted only for FT access.
+The CloudFormation template for the S3 bucket is [here](https://github.com/Financial-Times/content-k8s-provisioner/blob/master/helm-repo-stack/helm-s3-repo.yaml)
```
Comment (tamas-molnar) [inline Pipeline_integration_points.md]: Missing period at the end of the sentence.

File: Pipeline_integration_points.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+For uploading Helm chats to this [upp-helm-repo](https://s3.console.aws.amazon.com/s3/buckets/upp-helm-repo/?region=eu-west-1&tab=overview) S3 bucket, the pipeline uses the AWS user [upp-helm-repo-access.prod](https://console.aws.amazon.com/iam/home?region=eu-west-1#/users/upp-helm-repo-access.prod)
```
Comment (tamas-molnar) [inline Pipeline_integration_points.md]: Helm charts, not chats.

File: Test_pipeline_changes.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    1. In the **Pipeline libraries** part define the overwritting library
+    ![Add pipeline definition](https://user-images.githubusercontent.com/8848332/44658096-23177400-aa08-11e8-8607-af78a4288d06.png)
+    Use the following details for the library:
+          - Name: must be **k8s-pipeline-lib**
+          - Default version: the name of your feature multibranch
+          - Make sure you fill in the details of Git repository where the library code exists.
+    1. Save the job configuration
+3. Test that both the team envs & upper envs deploy work by either creating new temporary releases or triggering the existing jobs for current releases.
+4. After testing is done and everything is fine, please remove the Pipeline library from the configuration of the job.
+
```
Comment (tamas-molnar) [inline Test_pipeline_changes.md]: Link the "After testing pipeline changes" section here, not everyone will scroll down.

File: Test_pipeline_changes.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    2. Click on the "New Item" and input the following:
+![  copy](https://user-images.githubusercontent.com/8848332/44659662-a8515780-aa0d-11e8-8592-c215b67922cc.png)
+        - Copy from : **k8s-deployment/utils/deploy-upp-helm-chart**
+        - Item name: any name would do here
+    1. Afterwards you are taken to the configuration screen of the new job. You must adjust the version of the library that is used by putting the name of your branch in the script code:
+![script](https://user-images.githubusercontent.com/8848332/44659677-b43d1980-aa0d-11e8-8288-51c446d06bf3.png)
+    1. Click Save
+1. Manually trigger this new job for testing the introduced functionality
+2. After you're done with the testing and happy with the result, delete this job.
+
```
Comment (tamas-molnar) [inline Test_pipeline_changes.md]: Link the "After testing pipeline changes" section here, not everyone will scroll down.

File: README.md [hunk 5]
```diff
@@ -1,10 +1,23 @@
...
 
 ## Documentation
-[Deployment in k8s](https://docs.google.com/a/ft.com/document/d/15ecubJwkszH1B360Ah31uXy2UekpWlgfEmQeH9_wko8/edit?usp=sharing)
+[Deployment in k8s - spec](https://docs.google.com/document/d/1eNOczq8tEG8Q2boqKqjFKis9qMIRdQi6vDLgrWy4Akk)
+
+### Main global steps
+1. [Generic entry point for Jenkinsfile](vars/genericEntryPointForJenkinsfile.groovy) - the generic entry point for the continuous delivery pipelines. See [spec](https://docs.google.com/document/d/1eNOczq8tEG8Q2boqKqjFKis9qMIRdQi6vDLgrWy4Akk/edit#heading=h.3dr93sgi36h)
+1. [UPP entry point for Jenkinsfile](/vars/uppEntryPointForJenkinsfile.groovy) main entry point for Continuous Delivery in UPP clusters. See [spec](https://docs.google.com/document/d/1eNOczq8tEG8Q2boqKqjFKis9qMIRdQi6vDLgrWy4Akk/edit#heading=h.rcnsi4cl7nnz)
+1. [PAC entry point for Jenkinsfile](/vars/uppEntryPointForJenkinsfile.groovy) main entry point for Continuous Delivery in UPP clusters. See [spec](https://docs.google.com/document/d/1eNOczq8tEG8Q2boqKqjFKis9qMIRdQi6vDLgrWy4Akk/edit#heading=h.rcnsi4cl7nnz)
```
Comment (tamas-molnar) [inline README.md]: Links to upp entry point

File: README.md [hunk 6]
```diff
@@ -1,10 +1,23 @@
...
+
+### Utility global steps
+1. [Diff & sync 2 envs](vars/diffBetweenEnvs.groovy): - this is the main step used in the [Diff & Sync 2 k8s envs](https://upp-k8s-jenkins.in.ft.com/job/k8s-deployment/job/utils/job/diff-between-envs/). This job can be used to keep the team envs in sync with the prod ones, or when provisioning a new environment.
+1. [Update cluster using the provisioner](vars/updateCluster.groovy) - this is the main step used in the [Update a Kubernetes cluster using the Provisioner](https://upp-k8s-jenkins.in.ft.com/job/k8s-deployment/job/utils/job/update-cluster/). This job can be used for updating the CoreOS version in a cluster.
+1. [Update Dex configs](vars/updateDexConfigs.groovy) - this is the main step used in the job [Update Dex Config](https://upp-k8s-jenkins.in.ft.com/job/k8s-deployment/job/utils/job/update-dex-config/) that updates the Dex configurations in multiple clusters at once.
```
Comment (tamas-molnar) [inline README.md]: Link to our Dex repo(s)? What is Dex?

File: README.md [hunk 7]
```diff
@@ -54,4 +67,74 @@ For each of the API servers in the environment Jenkins needs 1 key in order to a
 1. After merge, add the new environment to the Jenkins jobs:
     1. [Deploys a helm chart from the upp repo](https://upp-k8s-jenkins.in.ft.com/job/k8s-deployment/job/utils/job/deploy-upp-helm-chart/)
     1. [Update a Kubernetes cluster](https://upp-k8s-jenkins.in.ft.com/job/k8s-deployment/job/utils/job/update-cluster/)
-    
+
+# Developer documentation
```
Comment (tamas-molnar) [inline README.md]: Maybe split this into another file?

Comment (sbuliarca) [reply README.md]: I thought it could stay here, as it is not that big, and for big topics it does contain subpages.

**Pull request merged**